### ISSUE TITLE
Mars Orbit Insertion

### ISSUE THREAD

**Issue by ChristopherRabotin** [ChristopherRabotin/smd#34]
Grand finale: perform injection to Mars from interplanetary trajectory. Once this is done, move onto final project.

**Comment by ChristopherRabotin** [comment]
Currently targeting Mars makes the vehicle arrive with a hyperbolic orbit in the SOI, so it just does a fly-by and fails to inject. A current test that I need to debug targets the orbit right after it leaves Mars, but that led to a weird propagation with a huge semi major axis. I didn't look into it at all yet, and this is what this issue is about.

**Comment by ChristopherRabotin** [comment]
Targeting the Mars orbit mostly works, but Naasz does not completely converge, and I don't know why. This is a similar issue to what happens in the tests when trying to fix the argument of perigee.

```
>>> # TARGET a=263771273.5 e=0.2504 i=23.992 Ω=358.323 ω=21.008 ν=175.310 λ=194.642 u=196.319"
>>> # REACHES a=263771300.1 e=0.2504 i=23.997 Ω=7.187 ω=335.456 ν=313.166 λ=295.809 u=288.622
```

**Comment by ChristopherRabotin** [comment]
Things to test
- [ ] Target Mars directly instead of an orbit about Mars and then perform `ToElliptical` ==> vehicle collides with planet, unsurprisingly
- [ ] Simulate in/out hyperbolic spiral around Mars to ensure those work just as well as around Earth
- [ ] Target hyperbolic orbit about Mars which was validated from above
- [ ] In the above, target specifically Mars leaving vehicle as if it were leave by loosing energy (change `ToXCentric` to support the direction of escape -- better yet: calculate which it is exactly and tune the departure orbit)

**Comment by ChristopherRabotin** [comment]
Most of the changes from the `issue-34` branch have been added to other branches. The `issue-34` should **not** be merged into `master` and should be considered obsolete code.
